main.tf

```diff
@@ -92,3 +100,57 @@ resource "kubernetes_namespace" "namespace" {
     labels = each.value.labels
   }
 }
+
+resource "kubernetes_network_policy" "isolated" {
+  for_each = {
+    for p in local.drupal_network_policy_list : p.namespace => p.project_name if p.network_policy == "isolated"
+  }
+
+  metadata {
+    name      = "network-policy-isolated-${each.value}"
+    namespace = each.key
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+  }
+
+  depends_on = [
+    kubernetes_namespace.namespace
+  ]
```

Comment:
I would prefer the `kubernetes.namespace[<key>].<namespace name argument>` in the `metadata.namespace` field to avoid explicit dependency.

What do you think?

Reply:
If I explicitly input "[each.key]", it gives me an error.
```
│ Error: Invalid expression
│ 
│   on .terraform/modules/drupal_resources_for_projects_in_the_shared_cluster/main.tf line 148, in resource "kubernetes_network_policy_v1" "restricted":
│  148:     kubernetes_namespace.namespace[each.key]
│ A single static variable reference is required: only attribute access and indexing with constant keys. No calculations, function
│ calls, template expressions, etc are allowed here.
```